botocore/parsers.py

```diff
@@ -386,10 +386,13 @@ def _do_error_parse(self, response, shape):
         root = self._parse_xml_string_to_dom(xml_contents)
         parsed = self._build_name_to_xml_node(root)
         self._replace_nodes(parsed)
-        # Once we've converted xml->dict, we need to make one more
-        # adjustment to be consistent with ResponseMetadata
-        # for non-error responses:
-        # {"RequestId": "id"} -> {"ResponseMetadata": {"RequestId": "id"}}
+        # Once we've converted xml->dict, we need to make one or two
+        # more adjustments to extract nested errors and to be consistent
+        # with ResponseMetadata for non-error responses:
+        # 1. {"Errors": {"Error": {...}}} -> {"Error": {...}}
+        # 2. {"RequestId": "id"} -> {"ResponseMetadata": {"RequestId": "id"}}
+        if 'Errors' in parsed and isinstance(parsed['Errors'], dict):
```

Comment:
I'm not able to find a unit test for this change.  If there's not one we should get one in place for this.

Also, why the isinstance check for dict?  Are there cases where there's an Errors key and it's not a dict?